src/stdio.rs

```diff
@@ -0,0 +1,243 @@
+use super::Error;
+
+use core::mem::ManuallyDrop;
+
+use core::pin::Pin;
+use core::task::{Context, Poll};
+
+use std::fs::File;
+use std::io::{self, IoSlice};
+use std::os::unix::io::{AsRawFd, FromRawFd, IntoRawFd, RawFd};
+
+use std::process;
+
+use tokio::io::{AsyncRead, AsyncReadExt, AsyncWrite, ReadBuf};
+
+#[derive(Debug)]
+pub(crate) enum StdioImpl {
+    /// Read/Write to /dev/null
+    Null,
+    /// Read/Write to a newly created pipe
+    Pipe,
+    /// Read/Write to custom fd
+    Fd(File),
+}
+
+/// Similar to std::process::Stdio
```

Comment:
I wonder how `tokio` gets away with not having its own `Stdio` type :thinking:

Reply:
Check out [this](https://docs.rs/tokio/1.13.0/src/tokio/process/mod.rs.html#222).

`tokio` uses `std::process::Command` internally to spawn a child, they then [take the `child.stdin`, `child.stdout` and `child.stderr](https://docs.rs/tokio/1.13.0/src/tokio/process/unix/mod.rs.html#102) and [convert them into `ChildStdio = PollEvented<Pipe>`](https://docs.rs/tokio/1.13.0/src/tokio/process/unix/mod.rs.html#241).

Reply:
Huh, neat. Thanks for the pointer!